grammars/prql-lezer/src/prql.grammar

```diff
@@ -74,7 +74,7 @@ Op_bin { op_bin_only | !term op_unary }
   line_wrap { "\\" }
   wrapped_line { newline+ (Comment newline+)* line_wrap }
   newline { "\n" }
-  Range { ".." }
+  Range { @digit+ ".." @digit+ }
```

Comment:
Because this can also be used to compile to `BETWEEN`, this can be any literal, and arguably any expression. It also shouldn't allow for spaces — is that possible with `![ ]`?

Could we test with `5..(-10)` and `@2022..@2023`?

(If some of these are very difficult, we can merge the partial version though)

Reply:
I had a look but you're right, it isn't trivial...

Reply:
Yeah, it seems we can use `Number` instead of `@digit+` but I don't know if that is desirable because then `123_456e10..123.00` would be valid which I am not sure it should be.

Reply:
It's much better to be too permissive than too strict here, because false positives will be quickly corrected by the user (because it's invalid PRQL), but false negatives will continue to exist (because it's valid PRQL)

Reply:
(that said, I'll merge and if you want we can make the change in another PR)